cmake/Modules/FindSYCLToolkit.cmake

```diff
@@ -1,54 +1,97 @@
-# This will define the following variables:
-# SYCL_FOUND               : True if the system has the SYCL library.
-# SYCL_INCLUDE_DIR         : Include directories needed to use SYCL.
-# SYCL_LIBRARY_DIR         ：The path to the SYCL library.
-# SYCL_LIBRARY             : SYCL library fullname.
+#[=======================================================================[.rst:
+SYCLConfig
+-------
 
-include(FindPackageHandleStandardArgs)
+Library to verify SYCL compatability of CMAKE_CXX_COMPILER
+and passes relevant compiler flags.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``SYCLTOOLKIT_FOUND``
+  True if the system has the SYCL library.
+``SYCL_COMPILER``
+  SYCL compiler executable.
+``SYCL_INCLUDE_DIR``
+  Include directories needed to use SYCL.
+``SYCL_LIBRARY_DIR``
+  Libaray directories needed to use SYCL.
+``SYCL_FLAGS``
+  SYCL specific flags for the compiler.
+``SYCL_LANGUAGE_VERSION``
+  The SYCL language spec version by Compiler.
+
+``SYCL::SYCL_CXX``
+  Interface target for using SYCL compiler.  The following properties are
+  defined for the target: ``INTERFACE_COMPILE_OPTIONS``,
+  ``INTERFACE_LINK_OPTIONS``, ``INTERFACE_INCLUDE_DIRECTORIES``, and
+  ``INTERFACE_LINK_DIRECTORIES``
+
+#]=======================================================================]
+
+set(SYCLTOOLKIT_FOUND False)
+include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
 
 set(SYCL_ROOT "")
 if(DEFINED ENV{SYCL_ROOT})
   set(SYCL_ROOT $ENV{SYCL_ROOT})
 elseif(DEFINED ENV{CMPLR_ROOT})
   set(SYCL_ROOT $ENV{CMPLR_ROOT})
 endif()
+if(NOT SYCL_ROOT)
+  execute_process(
+    COMMAND which icpx
+    OUTPUT_VARIABLE SYCL_CMPLR_FULL_PATH
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-string(COMPARE EQUAL "${SYCL_ROOT}" "" nosyclfound)
-if(nosyclfound)
-  set(SYCL_FOUND False)
-  set(SYCL_REASON_FAILURE "SYCL library not set!!")
+  if(NOT EXISTS "${SYCL_CMPLR_FULL_PATH}")
+    message("Cannot find ENV{CMPLR_ROOT} or icpx, please setup SYCL compiler Tool kit enviroment before building!!")
+    return()
+  endif()
+
+  get_filename_component(SYCL_BIN_DIR "${SYCL_CMPLR_FULL_PATH}" DIRECTORY)
+  set(SYCL_ROOT ${SYCL_BIN_DIR}/..)
+endif()
```

Comment:
We don't need to support this case. If SYCL_ROOT is not defined, we can raise error directly.

Reply:
2 ways to find SYCL_ROOT,
1. Find it basing on environment variables: 1) CMPLR_ROOT (oneAPI toolkit env), 2) SYCL_ROOT (we defined)
2. Find it basing on icpx installation path. It is aligned CUDA implementation, in case environment variables are polluted.

Reply:
Will we always recommend user to source the env.sh in oneAPI package?

Reply:
Yes, it's the first choice to setup SYCL compiler toolkit environment.

Reply:
Then, let's always assume the env setup should be ready by env.sh, and use all pre-defined macro in the env, instead of hard-code name, for example, "icpx"?
And, we should not support other customized setup here?

BTW, why we do not use  `find_package(IntelSYCL REQUIRED)`

Reply:
OK. We cannot use these utilities because we cannot allow to set default compiler as dpc++. Am I right?

Reply:
Right. So far, Intel SYCL FindModule cmake in oneAPI package assumes Intel SYCL compiler is CXX compiler. However, for separate compilation, it doesn't work. And I abstract what we need from the cmake.